### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by najibullahjafari**
Hello Code Reviewer!
Hope you are doing great. 🤓
In this Milstone,
1. Install rspec ( gem install rspec ) for Test Cases.
2. Write unit tests in pair-programming session for all the methods in all the classes:
- [x] Book
- [x] Person
- [x] Student
   Added by @najibullahjafari 
- [x] Teacher
- [x] Classroom
- [x] Rental
- [x] Nameable
   Added by @MasumaJaffery 
3. Run The Test Cases That shows us Pass Status.
Thanks for your Time to Review our PR.
Happy Reviewing!
Stay Blessed ✨️

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require_relative '../person'
+
+describe Person do
+  context 'test person class' do
```
Comment (Owusu-Desmond) [inline spec/person_spec.rb]: - [ ] Please ensure that you add a unit test for the `can_use_services?` method and make sure that you test positive and negative case scenarios. Kindly test the can_use_services?` method that returns true if has parent permission and returns false if under age and no parent permission.
- [ ] Make sure that you also have a unit test that tests the trimming of the person's name. Kindly add a test for the correct_name method that trims the name to 10 characters :pray:

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    it 'This test should return the exact data' do
+      person = Person.new('Najib', 22)
+      age = person.age
+      name = person.name
+      expect(age).to eq(22)
+      expect(name).to eq('Najib')
+    end
+  end
```
Comment (Owusu-Desmond) [inline spec/person_spec.rb:37]: - [ ] Please make sure that you also add a test to expect person rentals :pray: 
- [ ] Kindly ensure that you add a test to test adding new rentals. The add_rental method should return a rental and add it to the book.

File: spec/teacher_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require_relative '../teacher'
+
+describe Teacher do
```
Comment (Owusu-Desmond) [inline spec/teacher_spec.rb:3]: - [ ] I suggest you test the `can_use_services?` method in the teacher.rb class since all the methods in all the classes should be tested :blush:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @najibullahjafari and @MasumaJaffery , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    end
+
+    it "should add a rental to a person's rentals list" do
+      person = Person.new('Najib', 22, parent_permission: true)
+      book = Book.new('Book Title', 'Author')
+      date = Date.new(2023, 9, 30)
+      person.add_rental(book, date)
+      expect(person.rentals).to include(an_instance_of(Rental))
+    end
+  end
```
Comment (skyv26) [inline spec/person_spec.rb:37]: - [ ] Well done 👍🏻 @najibullahjafari and @MasumaJaffery , you both did a great job by implementing the changes asked by your previous code reviewer. Even the things seem correct when I run the test for the `spec` folder. But I tried to run to the `individual` file i.e. `person_spec` file. Your one test is failing the test just because you didn't add the `Date` library and because of that I got below given screenshot error. 

Screenshot from 2023-09-22 04-52-42


### Solution

**Please add the `require 'date'` at the top of the file and your problem will be resolved, if you want to verify that you can use the below command** 
```sh 
rspec spec/person_spec.rb
```

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @najibullahjafari and @MasumaJaffery, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Implemented the features asked by your previous code reviewer ✅ 
- Professional `README` template ✅ 
- Well-structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- Please make sure that your PR description is descriptive enough, which will help the person understand what you implemented in your feature branch. Right now your PR description does contain the fact about containing tests for the listed files, but you can also consider adding about what you did thoroughly, even can add related to what you achieved or learned as an individual.

Screenshot from 2023-09-22 04-55-45


#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbayomiOlaoye — APPROVED** [review]
## STATUS: APPROVED :green_circle:  

Hi @Team  :wave: :wave:, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

applause

## Highlights
- Cleanly written code :white_check_mark: 
- App runs well and functions as expected ✅ 
- Nicely implemented test cases ✅ 
- Great attitude to feedback ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
